common/types/hashable.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+// A chain-agnostic generic interface to represent the following native types on various chains:
+// PublicKey, Address, Account, BlockHash, TxHash
+type Hashable interface {
```

Comment:
nit: to me `Hashable` implies that the underlying struct can be passed to a hash func or that the methods provide hashing funcs

this interface feels so generic that the naming shouldn't be constrained just to blockchain concepts

Reply:
Maybe `ByteStringer` or `StringByteable`, especially if we are going to use the interfaces defined in https://pkg.go.dev/encoding ?

Reply:
I'm open. Will prefer to do the large rename as per everyone's opinion.

Reply:
Yes, I think unique identifier is the closest thing that matches these.
So Uniques?